prow/plugins/require-matching-label/require-matching-label.go

```diff
@@ -0,0 +1,213 @@
+		repo:          ie.Repo.Name,
+		number:        ie.Issue.Number,
+		author:        ie.Issue.User.Login,
+		label:         ie.Label.Name, // This will be empty for non-label events.
+		currentLabels: ie.Issue.Labels,
+	}
+	return handle(pc.Logger, pc.GitHubClient, pc.CommentPruner, pc.PluginConfig.RequireMatchingLabel, e)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pre github.PullRequestEvent) error {
+	if !handlePRActions[pre.Action] {
+		return nil
+	}
+	e := &event{
+		org:    pre.Repo.Owner.Login,
+		repo:   pre.Repo.Name,
+		number: pre.PullRequest.Number,
+		author: pre.PullRequest.User.Login,
+		label:  pre.Label.Name, // This will be empty for non-label events.
+	}
+	return handle(pc.Logger, pc.GitHubClient, pc.CommentPruner, pc.PluginConfig.RequireMatchingLabel, e)
+}
+
+// matchingConfigs filters irrelevant RequireMtchingLabel configs from
+// the list of all configs.
+// `branch` should be empty for Issues and non-empty for PRs.
+// `label` should be omitted in the case of 'open' and 'reopen' actions.
+func matchingConfigs(org, repo, branch, label string, allConfigs []plugins.RequireMatchingLabel) []plugins.RequireMatchingLabel {
+	var filtered []plugins.RequireMatchingLabel
+	for _, cfg := range allConfigs {
+		// Check if the config applies to this issue type.
+		if (branch == "" && !cfg.Issues) || (branch != "" && !cfg.PRs) {
+			continue
+		}
+		// Check if the config applies to this 'org[/repo][/branch]'.
+		if org != cfg.Org ||
+			(cfg.Repo != "" && cfg.Repo != repo) ||
+			(cfg.Branch != "" && branch != "" && cfg.Branch != branch) {
+			continue
+		}
+		// If we are reacting to a label event, see if it is relevant.
+		if label != "" && !cfg.Re.MatchString(label) {
+			continue
+		}
+		filtered = append(filtered, cfg)
+	}
+	return filtered
+}
+
+func handle(log *logrus.Entry, ghc githubClient, cp commentPruner, configs []plugins.RequireMatchingLabel, e *event) error {
+	// Find any configs that may be relevant to this event.
+	matchConfigs := matchingConfigs(e.org, e.repo, e.branch, e.label, configs)
+	if len(matchConfigs) == 0 {
+		return nil
+	}
+
+	if e.label == "" /* not a label event */ {
+		// If we are reacting to a PR or Issue being created or reopened, we should wait a
+		// few seconds to allow other automation to apply labels in order to minimize thrashing.
+		sleep(time.Second * 5)
+		// If currentLabels was populated it is now stale.
+		e.currentLabels = nil
+	}
+	if e.currentLabels == nil {
+		var err error
+		e.currentLabels, err = ghc.GetIssueLabels(e.org, e.repo, e.number)
+		if err != nil {
+			return fmt.Errorf("error getting the issue or pr's labels: %v", err)
+		}
+	}
+
+	// Handle the potentially relevant configs.
+	for _, cfg := range matchConfigs {
+		hasMissingLabel := false
+		hasMatchingLabel := false
+		for _, label := range e.currentLabels {
+			hasMissingLabel = hasMissingLabel || label.Name == cfg.MissingLabel
+			hasMatchingLabel = hasMatchingLabel || cfg.Re.MatchString(label.Name)
```

Comment:
We should validate that `cfg.Re.MatchString(cfg.MissingLabel) == false`

Reply:
Done.